### ISSUE TITLE
Release artifacts (models, dataset) for Learning To Defer to a Population on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [nil123532/learning-to-defer-to-a-population-with-limited-demonstrations#1]
Hi @nil123532 🤗

Niels here from the open-source team at Hugging Face. I discovered your work on Arxiv and its associated GitHub repository: `https://github.com/nil123532/learning-to-defer-to-a-population-with-limited-demonstrations`.
Your paper, "Learning To Defer To A Population With Limited Demonstrations" (https://huggingface.co/papers/2510.19351), introduces an interesting framework for L2D systems and the generation of pseudo-labels.

I noticed that while you've provided the code and configurations to run your experiments and generate the pseudo-labels and train models, the pre-trained model checkpoints (e.g., the Embedding Model, L2D-POP models) and the generated pseudo-label datasets themselves are not directly available for download.

It'd be great to make these artifacts (pre-trained model checkpoints and/or generated pseudo-label datasets) available on the 🤗 hub, to improve their discoverability and visibility. We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

## Uploading dataset (pseudo-labels)

Would be awesome to make the generated pseudo-label datasets available on 🤗 , so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗
